### PR TITLE
feat: address + discount

### PR DESCRIPTION
**Pull request by nayyyhaa**
## In this PR :
-  I've added discount + address popup.

#### Component files

- [x] [components/AddressPopup.js](https://github.com/nayyyhaa/inNotion-care-react/compare/payment-branch?expand=1#diff-0f83652e76fab191a0c39c77dd5b0b638f4d26895ed0366bd9b3f876a93628c2)
- [x] [components/CartSidebar.js](https://github.com/nayyyhaa/inNotion-care-react/compare/payment-branch?expand=1#diff-bc3b67a94629380a52356fc4046c3c92e564e1634d523d9a0620c6c53d76c31c)
- [x] [components/DiscountPopup.js](https://github.com/nayyyhaa/inNotion-care-react/compare/payment-branch?expand=1#diff-ce1a3b7a30b2adcaad26968d06331096c32bbe2ffde73d6247ed1b9a874f108f)

#### Page
- [x] [pages/CartPage.js](https://github.com/nayyyhaa/inNotion-care-react/compare/payment-branch?expand=1#diff-9f61638ebeb4120ea15dc0d5bbc4610f323139fc0465bc2b6600716dcc373582)
- [x] [pages/CheckoutPage.js](https://github.com/nayyyhaa/inNotion-care-react/compare/payment-branch?expand=1#diff-8c5148e4fe86886ce753a8cedea65f09a41a1c5a7912363f2866d41575dfc3be)

#### utils:
- [x] [contexts/AddressContext.js](https://github.com/nayyyhaa/inNotion-care-react/compare/payment-branch?expand=1#diff-078a9105f62278d458dc210111732c7763be50555480171ad929f5cc940f711c)
 
#### files to *ignore*
- [ ] files with just few import changed
- [ ] CSS files

### Live preview : https://deploy-preview-8--innotion-carev2.netlify.app/

### CODE REVIEW

File: src/pages/CartPage.js [hunk 1]
```diff
@@ -1,32 +1,51 @@
...
+                Checkout our{" "}
+                <Link to="/products" className="colored-text">
+                  products
+                </Link>
+              </p>
+            </div>
+          )}
+          {/* {cart?.map((product) => (
+            <CartCard key={product._id} product={product} />
+          ))} */}
```
Comment (vrinda-mahajan) [inline src/pages/CartPage.js]: You can remove the commented code if not required.

File: src/pages/CheckoutPage.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+                user?.email === data?.email && (
+                  <li key={data?._id} className="filter-item address-box m-v-1">
+                    <label className="categories cursor">
+                      <input
+                        type="radio"
+                        className="radio-input m-v-05 m-r-1"
+                        name="address"
+                        id={data?._id}
+                        // checked={data?.code === code}
+                        // onChange={() => setCode(data?.code)}
```
Comment (vrinda-mahajan) [inline src/pages/CheckoutPage.js:46]: Here also remove the comments.

**Review by vrinda-mahajan — APPROVED** [review]
The code looks clean and the functionality is also working perfectly. You can merge the PR

**Review by arjun-computer-geek — APPROVED** [review]
code is well formatted. good class name and all functionality are working..

**Comment by nayyyhaa** [comment]
Thanks @vrinda-mahajan & @arjun-computer-geek for review. Merging the changes.

**Pull request merged**